client/homescreen/activity-panel/reviews/index.js

```diff
@@ -0,0 +1,436 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import classnames from 'classnames';
+import { Component, Fragment } from '@wordpress/element';
+import { Button } from '@wordpress/components';
+import { compose } from '@wordpress/compose';
+import { withSelect, withDispatch } from '@wordpress/data';
+import PropTypes from 'prop-types';
+import CheckmarkIcon from 'gridicons/dist/checkmark';
+import StarIcon from 'gridicons/dist/star';
+import StarOutlineIcon from 'gridicons/dist/star-outline';
+import interpolateComponents from 'interpolate-components';
+import {
+	EmptyContent,
+	Link,
+	ReviewRating,
+	ProductImage,
+	Section,
+} from '@woocommerce/components';
+import { getAdminLink } from '@woocommerce/wc-admin-settings';
+import { get, isNull } from 'lodash';
+import { REVIEWS_STORE_NAME } from '@woocommerce/data';
+import { recordEvent } from '@woocommerce/tracks';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import {
+	ActivityCard,
+	ActivityCardPlaceholder,
+} from '../../../header/activity-panel/activity-card';
+import { CurrencyContext } from '../../../lib/currency-context';
+import sanitizeHTML from '../../../lib/sanitize-html';
+import { REVIEW_PAGE_LIMIT } from './utils';
+
+class ReviewsPanel extends Component {
+	recordReviewEvent( eventName, eventData ) {
+		recordEvent( `reviews_${ eventName }`, eventData || {} );
+	}
+
+	deleteReview( reviewId ) {
+		const {
+			deleteReview,
+			createNotice,
+			updateReview,
+			clearReviewsCache,
+			reviews,
+		} = this.props;
+		if ( reviewId ) {
+			deleteReview( reviewId )
+				.then( () => {
+					clearReviewsCache();
+					createNotice(
+						'success',
+						__(
+							'Review successfully deleted.',
+							'woocommerce-admin'
+						),
+						{
+							actions: [
+								{
+									label: __( 'Undo', 'woocommerce-admin' ),
+									onClick: () => {
+										updateReview(
+											reviewId,
+											{
+												status: 'untrash',
+											},
+											{
+												_embed: 1,
+											}
+										).then( () =>
+											clearReviewsCache(
+												reviews.length < 2
```

Comment:
Why is this `2`? Doesn't the panel show `5` reviews?

Reply:
Yeah it shows 5, this was in reference to this change: https://github.com/woocommerce/woocommerce-admin/pull/5706#discussion_r528972306 to only call it if it was the last item being updated.
I could probably check if `reviews.length` would actually be zero (I didn't specifically test that, as I thought it might show as `1` still within the scope).